src/components/Project/ImageAnalysis/ImageViewer.vue

```diff
@@ -164,6 +130,17 @@ onMounted(() => {
     })
   })
 
+  // Get coordinates of the line
+  image.on('pm:create', (e) => {
+    // Save last drawn line
+    lastDrawnLine = e.layer
+    e.layer.on('pm:edit', handleEdit)
+    findPixelCoords(e.layer.getLatLngs())
+  })
+}
+
+// Callback function for when a line is edited and checks if it's within bounds
+function handleEdit(event) {
   // Checks if the point is within the bounds of the image
   function isWithinBounds(point) {
```

Comment:
this being inside the `handleEdit` does it get triggered when the user first draws a line? In other words, if the user draws a line and extends it outside of the image, does the endpoint stay within bounds?

Reply:
I moved it outside at Matt's request and tested it again and looks like it is working great, The point is staying inside the bounds of the image even when the user tries to draw it outside.